tree/dataframe/test/dataframe_concurrency.cxx

```diff
@@ -234,5 +243,32 @@ TEST(RDFConcurrency, ParallelRDFCachesEnableImplicitMT)
    ParallelRDFCaches();
    ROOT::DisableImplicitMT();
 }
-
 #endif
+
+// Check that multiple RDF can JIT at the same time without interfering
+// with each other
+TEST(RDFConcurrency, JITWithManyThreads)
+{
+   ROOT::EnableThreadSafety();
+
+   std::vector<int> expected(25);
+   for (int i = 0; i < 25; i++)
+      expected[i] = i;
+
+   std::vector<int> results(25);
+   auto do_work = [&results](int slot) {
+      for (int i = slot * 5; i < (slot * 5 + 5); i++) {
+         results[i] = ROOT::RDataFrame{1}.Define("x", std::to_string(i)).Sum<int>("x").GetValue();
```

Comment:
This test currently has random behavior (this is actually a good thing and we should keep as is but might want to add more tests) and thus in the extreme can spuriously succeed for example if the system choose to serialize all the threads.  We should probably add another version of the test that insure the ordering we are looking for (by using some sort of barriers, for example 'spin locks' over an atomic).

Reply:
I do not believe that is possible with the existing RDataFrame API. Since there is no way to call `RLoopManager::Jit`, there is no way for me to synchronize the jitting stages of the two RDFs. The only thing I can use as a synchronization trigger is the trigger of the event loop, but that means doing everything, calling `RLoopManager::Jit`, calling `RDFInternal::InterpreterCalc`. At that point, there is no way the second thread will have had time to reach the Jit function and take an empty string because it will have been blocked by the first thread's event loop and thus when it reaches the Jit function it just sees the string already empty and early returns

Reply:
> At that point, there is no way the second thread will have had time to reach the Jit function and take an empty string because it will have been blocked by the first thread's event loop and thus when it reaches the Jit function it just sees the string already empty and early returns

You indeed right.